libqtile/widget/systray.py

```diff
@@ -110,6 +110,7 @@ def __init__(self, **config):
         self.add_defaults(Systray.defaults)
         self.icons = {}
         self.screen = 0
+        self.tray_configured = False
```

Comment:
I think you can use `self.configured` already, as this is part of `_Widget` and is set to `True` the very first time `_Widget._configure` is called.

Reply:
I thought about that but wouldn't `self.configured` be set after this line `base._Widget._configure(self, qtile, bar)`? In that case, all that X window stuff wouldn't even run the first time. That's why I used a different variable.

Reply:
Alternatively, we could move `base._Widget._configure(self, qtile, bar)` to the end of `_configure`. I don't think that would cause too many issues.

Reply:
Oh yeah! I think instead we should set it to `True` within `bar.py` then, when the bar actually does the configuring. This way, all levels of `_configure` across all classes on widgets will have configured correctly set at all times.

Reply:
Sorry - not sure I understand what you're proposing there. Can you explain again?

Reply:
We can stick `widget.configured = True` after this line: https://github.com/qtile/qtile/blob/master/libqtile/bar.py#L253

Reply:
Ah, I see. And remove this line: https://github.com/qtile/qtile/blob/master/libqtile/widget/base.py#L204 ?

I can put that in a separate PR as it would prevent the need for the hack here.

Reply:
Yeah that should work I think. Maybe stick it in a commit before this commit here, might be easier to then check it's working fine in the systray widget (updated to use `self.configured` obvs)

Reply:
I've put that commit in separately (and pulled it out into it's own PR)